pkg/btcdiff/chain.go

```diff
@@ -0,0 +1,41 @@
+package btcdiff
```

Comment:
What's the point of the `pkg/btcdiff` package? Can't we just have the `BitcoinDifficultyChain` interface right in the `pkg/maintainer`?

Reply:
The idea is that we treat the `BitcoinDifficultyChain` as a separate chain, just like `BeaconChain` and `TbtcChain`.
Since the two have their interfaces defined in `pkg/beacon` and `pkg/tbtc` it seemed reasonable to define one for `BitcoinDifficultyChain` (`pkg/btcdiff`) as well. 

But since it is indeed not used anywhere, I deleted it the whole `btcdiff` directory in 6bd38d7bdc5e20bd5adda825fd08df472f14d4c3.